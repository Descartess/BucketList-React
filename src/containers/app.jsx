import React, { Component } from 'react';
import Banner from '../components/banner';

import SignInForm from '../components/signinform';
import SignUpForm from '../components/signupform';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showSignIn: false,
      showSignUp: false,
      username: '',
      password: '',
      rpassword: '',

    };
    this.onSignInShow = this.onSignInShow.bind(this);
    this.onSignUpShow = this.onSignUpShow.bind(this);
    this.handleUsernameInput = this.handleUsernameInput.bind(this);
    this.handleRepeatPasswordInput = this.handleRepeatPasswordInput.bind(this);
    this.handlePasswordInput = this.handlePasswordInput.bind(this);
  }
  onSignInShow() {
    this.setState({ showSignIn: !this.state.showSignIn });
  }

  onSignUpShow() {
    this.setState({ showSignUp: !this.state.showSignUp });
  }

  handleUsernameInput(e) {
    this.setState({ username: e.target.value });
  }

  handlePasswordInput(e) {
    this.setState({ password: e.target.value });
  }

  handleRepeatPasswordInput(e) {
    this.setState({ rpassword: e.target.value });
  }

  render() {
    console.log(this.state);
    return (
      <div>
        <Banner
          onSignIn={this.onSignInShow}
          onSignUp={this.onSignUpShow}
        />
        <SignInForm
          data={this.state}
          show={this.state.showSignIn}
          onSignIn={this.onSignInShow}
          userChange={this.handleUsernameInput}
          passwordChange={this.handlePasswordInput}
        />
        <SignUpForm
          data={this.state}
          show={this.state.showSignUp}
          onSignUp={this.onSignUpShow}
          userChange={this.handleUsernameInput}
          passwordChange={this.handlePasswordInput}
          rpasswordChange={this.handleRepeatPasswordInput}
        />
      </div>
    );
  }
}

export default App;
