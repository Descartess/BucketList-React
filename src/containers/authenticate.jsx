import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { registerUser, signinUser } from '../actions';
import Banner from '../components/banner';

import SignInForm from '../components/signinform';
import SignUpForm from '../components/signupform';

class Authenticate extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showSignIn: false,
      showSignUp: false,
      username: '',
      password: '',
      rpassword: '',
    };
    this.onSignInShow = this.onSignInShow.bind(this);
    this.onSignUpShow = this.onSignUpShow.bind(this);
    this.handleUsernameInput = this.handleUsernameInput.bind(this);
    this.handleRepeatPasswordInput = this.handleRepeatPasswordInput.bind(this);
    this.handlePasswordInput = this.handlePasswordInput.bind(this);
    this.onSignIn = this.onSignIn.bind(this);
    this.onSignUp = this.onSignUp.bind(this);
  }
  onSignInShow() {
    this.setState({ showSignIn: !this.state.showSignIn });
  }

  onSignIn() {
    const { username, password } = this.state;
    this.props.signinUser({ username, password });
  }

  onSignUp() {
    const { username, password } = this.state;
    this.props.registerUser({ username, password });
  }

  onSignUpShow() {
    this.setState({ showSignUp: !this.state.showSignUp });
  }

  handleUsernameInput(e) {
    this.setState({ username: e.target.value });
  }

  handlePasswordInput(e) {
    this.setState({ password: e.target.value });
  }

  handleRepeatPasswordInput(e) {
    this.setState({ rpassword: e.target.value });
  }

  render() {
    return (
      <div>
        <Banner
          onSignInShow={this.onSignInShow}
          onSignUpShow={this.onSignUpShow}
        />
        <SignInForm
          data={this.state}
          show={this.state.showSignIn}
          onSignIn={this.onSignIn}
          onSignInShow={this.onSignInShow}
          userChange={this.handleUsernameInput}
          passwordChange={this.handlePasswordInput}
        />
        <SignUpForm
          data={this.state}
          show={this.state.showSignUp}
          onSignUp={this.onSignUp}
          onSignUpShow={this.onSignUpShow}
          userChange={this.handleUsernameInput}
          passwordChange={this.handlePasswordInput}
          rpasswordChange={this.handleRepeatPasswordInput}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { loading } = state.auth;
  return { loading };
};

// Authenticate.propTypes = {
//   loading: PropTypes.bool.isRequired,
//   registerUser: PropTypes.func.isRequired,
//   signinUser: PropTypes.func.isRequired,
// };

export default connect(mapStateToProps, { registerUser, signinUser })(Authenticate);

export { Authenticate };
