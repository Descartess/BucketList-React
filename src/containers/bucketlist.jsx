import React, { Component } from 'react';
import { connect } from 'react-redux';
import Header from '../components/bucketlist_header';
import BucketListDetail from '../components/bucketlistdetail';
import AddBucketListForm from '../components/addbucketlist';
import ConfirmDelete from '../components/confirm_delete';
import EditBucketListForm from '../components/editbucketlist';
import { getBucketLists, postBucketList } from '../actions';

class BucketListContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedList: null,
      showAddBucketList: false,
      showEditBucketList: false,
      showDeleteBucketList: false,
      bucketlist_name: '',
      bucketlist_age: '',
    };
    this.showAddBucketList = this.showAddBucketList.bind(this);
    this.changeBucketListname = this.changeBucketListname.bind(this);
    this.changeBucketListage = this.changeBucketListage.bind(this);
    this.showDeleteBucketList = this.showDeleteBucketList.bind(this);
    this.showEditBucketList = this.showEditBucketList.bind(this);
  }
  componentWillMount() {
    this.props.getBucketLists();
  }
  changeBucketListname(e) {
    this.setState({ bucketlist_name: e.target.value });
  }
  changeBucketListage(e) {
    this.setState({ bucketlist_age: e.target.value });
  }
  showAddBucketList() {
    this.setState({ showAddBucketList: !this.state.showAddBucketList });
  }
  showEditBucketList() {
    this.setState({ showEditBucketList: !this.state.showEditBucketList });
  }
  showDeleteBucketList() {
    this.setState({ showDeleteBucketList: !this.state.showDeleteBucketList });
  }
  addBucketList() {
    const { bucketlist_name, bucketlist_age } = this.state;
    this.props.postBucketList({ name: bucketlist_name, completed_by: bucketlist_age });
  }
  renderBucketLists() {
    const { bucketlists } = this.props;
    if (bucketlists.length < 1) {
      return (
        <p> No bucketlists created.</p>
      );
    }
    return bucketlists.map(bucketlist => (
      <BucketListDetail
        key={bucketlist.id}
        name={bucketlist.name}
        delete={this.showDeleteBucketList}
        edit={this.showEditBucketList}
      />
      ),
    );
  }

  render() {
    return (
      <div>
        <div className="col-md-4">
          <div className="panel panel-default">
            <Header
              showAddBucketList={this.showAddBucketList}
            />
            <div className="panel-body">
              {this.renderBucketLists()}
            </div>
          </div>
        </div>
        <AddBucketListForm
          show={this.state.showAddBucketList}
          data={this.state}
          showAddBucketList={this.showAddBucketList}
          onNameChange={this.changeBucketListname}
          onAgeChange={this.changeBucketListage}
          addBucketList={this.addBucketList}
        />
        <EditBucketListForm
          show={this.state.showEditBucketList}
          data={this.state}
          showEditBucketList={this.showEditBucketList}
          onNameChange={this.changeBucketListname}
          onAgeChange={this.changeBucketListage}
          addBucketList={this.addBucketList}
        />
        <ConfirmDelete
          show={this.state.showDeleteBucketList}
          name="BucketList"
          cancel={this.showDeleteBucketList}
        />
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  const { bucketlists } = state.bucketlists;
  return { bucketlists };
};

export default connect(mapStateToProps,
  { getBucketLists, postBucketList })(BucketListContainer);
