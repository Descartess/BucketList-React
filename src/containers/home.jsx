import React, { Component } from 'react';
import { connect } from 'react-redux';
import Header from '../components/nav_header';
import { resetpassword, signout } from '../actions';
import ResetPassword from '../components/resetpassword';
import BucketListContainer from './bucketlist';
import ItemContainer from './items';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      old_password: '',
      new_password: '',
      showResetPassword: false,
    };
    this.showResetPassword = this.showResetPassword.bind(this);
    this.handleSelect = this.handleSelect.bind(this);
    this.passwordreset = this.passwordreset.bind(this);
    this.changeOldPassword = this.changeOldPassword.bind(this);
    this.changeUsername = this.changeUsername.bind(this);
    this.changeNewPassword = this.changeNewPassword.bind(this);
  }
  showResetPassword() {
    this.setState({ showResetPassword: !this.state.showResetPassword });
  }
  changeOldPassword(e) {
    this.setState({ old_password: e.target.value });
  }
  changeNewPassword(e) {
    this.setState({ new_password: e.target.value });
  }
  changeUsername(e) {
    this.setState({ username: e.target.value });
  }
  passwordreset() {
    const { username, old_password, new_password } = this.state;
    this.props.resetpassword({ username, old_password, new_password });
    this.showResetPassword();
  }
  handleSelect(eventKey) {
    if (eventKey === 1.2) {
      this.showResetPassword();
    } else if (eventKey === 1.1) {
      this.props.signout();
    }
  }
  render() {
    return (
      <div>
        <Header
          handleSelect={this.handleSelect}
        />
        <ResetPassword
          showResetPassword={this.showResetPassword}
          show={this.state.showResetPassword}
          passwordreset={this.passwordreset}
          changeUsername={this.changeUsername}
          changeNewPassword={this.changeNewPassword}
          changeOldPassword={this.changeOldPassword}
          data={this.state}
        />
        <BucketListContainer />
        <ItemContainer />
      </div>

    );
  }
}
const mapStateToProps = (state) => {
  const { bucketlists } = state.bucketlists;
  return { bucketlists };
};

export default connect(mapStateToProps, { resetpassword, signout })(Home);

export { Home };
