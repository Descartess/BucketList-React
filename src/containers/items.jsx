import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import Header from '../components/header';
import AddBucketListItemForm from '../components/additem';
import ItemDetail from '../components/item_detail';
import EditBucketListItem from '../components/editItem';
import ConfirmDelete from '../components/confirm_delete';
import { addBucketListItem,
  editBucketListItem,
  deleteBucketListItem,
  selectBucketItem,
} from '../actions';

class ItemContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showEditItem: false,
      showAddItem: false,
      showDeleteItem: false,
      item_name: '',
      completed: false,
    };
    this.renderItems = this.renderItems.bind(this);
    this.nameChange = this.nameChange.bind(this);
    this.showAddItem = this.showAddItem.bind(this);
    this.submitBucketListItem = this.submitBucketListItem.bind(this);
    this.itemsDisplay = this.itemsDisplay.bind(this);
    this.showDelete = this.showDelete.bind(this);
    this.showEditItem = this.showEditItem.bind(this);
    this.onToggle = this.onToggle.bind(this);
    this.editBucketItem = this.editBucketItem.bind(this);
  }
  showAddItem() {
    this.setState({ showAddItem: !this.state.showAddItem });
  }
  onToggle() {
    this.setState({ completed: !this.state.completed });
  }
  nameChange(e) {
    this.setState({ item_name: e.target.value });
  }
  submitBucketListItem() {
    const { item_name } = this.state;
    const { selectedBucket } = this.props;
    this.props.addBucketListItem(selectedBucket, { name: item_name });
  }
  showEditItem(item) {
    console.log(item);
    this.setState({
      item_name: item.name,
      completed: item.completed,
      showEditItem: !this.state.showEditItem,
    });
  }
  showDelete() {
    this.setState({ showDeleteItem: !this.state.showDeleteItem });
  }
  editBucketItem() {
    const { completed, item_name } = this.state;
    const { selectedBucket, selected_item } = this.props;
    this.props.editBucketListItem(selectedBucket, selected_item, {
      name: item_name,
      completed,
    });
  }
  itemsDisplay(bucket) {
    const { items } = bucket;
    if (items.length < 1) {
      return (
        <div> No items in this Bucketlist</div>
      );
    }
    return items.map(item => (
      <ItemDetail
        key={item.id}
        {...item}
        onSelect={() => this.props.selectBucketItem(item.id)}
        edit={() => this.showEditItem(item)}
        delete={this.showDelete}
      />
    ),
    );
  }
  renderItems() {
    const { selectedBucket, bucketlists } = this.props;
    if (selectedBucket) {
      const bucket = _.find(bucketlists, n => n.id === selectedBucket);
      return (
        <div className="panel panel-default">
          <Header
            title={bucket.name}
            add={this.showAddItem}
          />
          <div className="panel-body">
            {this.itemsDisplay(bucket)}
          </div>
        </div>);
    }
    return (
      <div className="panel panel-default">
        <Header
          title="Items"
        />
        <div className="panel-body">
          <p> Click on bucketlist to display items </p>
        </div>
      </div>);
  }
  render() {
    const { selectedBucket, selected_item } = this.props;
    return (
      <div>
        <div className="col-md-6">
          {this.renderItems()}
        </div>
        <AddBucketListItemForm
          show={this.state.showAddItem}
          onNameChange={this.nameChange}
          {...this.state}
          close={this.showAddItem}
          addBucketListItem={this.submitBucketListItem}
        />
        <ConfirmDelete
          show={this.state.showDeleteItem}
          name="BucketList Item"
          cancel={this.showDelete}
          delete={() => this.props.deleteBucketListItem(selectedBucket, selected_item)}
        />
        <EditBucketListItem
          {...this.state}
          close={() => this.setState({ showEditItem: !this.state.showEditItem })}
          show={this.state.showEditItem}
          onNameChange={this.nameChange}
          onToggle={this.onToggle}
          editBucketItem={this.editBucketItem}
        />
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  const { bucketlists, selectedBucket, selected_item } = state.bucketlists;
  return { bucketlists, selectedBucket, selected_item };
};
export default connect(mapStateToProps,
  { addBucketListItem,
    editBucketListItem,
    deleteBucketListItem,
    selectBucketItem,
  })(ItemContainer);

export { ItemContainer };
