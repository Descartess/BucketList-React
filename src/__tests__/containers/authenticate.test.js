import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Authenticate } from '../../containers/authenticate';

describe('<Authenticate />', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(
      <Authenticate />,
      );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('updates state when sign in button is clicked', () => {
    const component = mount(<Authenticate />);
    expect(component.state().showSignIn).toBeFalsy();
    component.find('button').at(0).simulate('click');
    expect(component.state().showSignIn).toBeTruthy();
  });
  it('updates state when sign up button is clicked', () => {
    const component = mount(<Authenticate />);
    expect(component.state().showSignUp).toBeFalsy();
    component.instance().onSignUpShow();
    expect(component.state().showSignUp).toBeTruthy();
  });
});
