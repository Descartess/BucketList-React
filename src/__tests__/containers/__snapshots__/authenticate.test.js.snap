// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authenticate /> renders correctly 1`] = `
<div>
  <div
    className="jumbotron"
  >
    <div
      className="row"
    >
      <div
        className="col-md-8"
      >
        <h1>
          BucketList
        </h1>
        <p>
          Track, Record and Share things you want to do before a certain age.
        </p>
      </div>
      <div
        className="col-md-4"
      >
        <button
          className="btn btn-lg btn-primary btn-block"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Sign In
        </button>
        <button
          className="btn btn-lg btn-success btn-block"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Sign Up
        </button>
      </div>
    </div>
  </div>
</div>
`;
